server/app/rest_api/rest.py

```diff
@@ -392,24 +441,24 @@ def post(self):
         try:
```

Comment:
Shouldn't these optional routes check to see if they are implemented by the engine, and return a 501 if not?

I would assume the engine would fill out the "features" implemented (as defined by driver), and that would then let the REST layer know when to immediately return a 501...

Not a blocker for the current release, but it seems like this file "assumes" that diffexp is *always* implemented.

Put another way - the engine should declare what it implements, and the REST layer should comply with that.

Reply:
I'm postponing this until I do the plugin architecture for the algorithms. It seemed like it wasn't going to be necessary for the initial release, and that it would be super clear how to do this with plugin architecture -- If there are no options for an algorithm then return 501.

Reply:
filed as issue #302